bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/binding/internal/servlet/MagicServlet.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.magic.binding.internal.servlet;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpContext;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Registers itself under "/" and provides links to other web UIs
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+@Component(immediate = true)
+public class MagicServlet implements Servlet {
+
+    protected HttpService httpService;
+
+    /** the root path of this web application */
+    public static final String WEBAPP_ALIAS = "/";
+
+    private final Logger logger = LoggerFactory.getLogger(MagicServlet.class);
+
+    @Activate
+    protected void activate(Map<String, Object> configProps, BundleContext bundleContext) {
+        try {
+            httpService.registerResources(WEBAPP_ALIAS, "web", new MagicHttpContext(bundleContext.getBundle()));
```

Comment:
Do we need a custom HTTP context for the resources of the magic bundle, too?
IIRC it has been used because of a buggy Eclipse implementation if fragments are used?
Is the fragment support here important?

Reply:
Is "web" a good url? Sounds pretty generic.

Reply:
I am registering everything under "/" now and jetty picks the context which has the closest match, i.e. it trys the full path and only falls back to "/" if no other servlet was responsible.

Reply:
The custom context is for returning the index page in case the user visits "/" without anything behind, so the index.html is shown to him.